src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/PackageSearchMetadataCache.cs

```diff
@@ -1,17 +1,35 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using NuGet.Protocol.Core.Types;
 
 namespace NuGet.PackageManagement.VisualStudio
 {
-    public class PackageSearchMetadataCache
+    public class PackageSearchMetadataCache : IDisposable
```

Comment:
This class hasn't been modified, except to make it disposable (no `IDisposable` fields or properties were added).  What's the purpose of making this disposable?  If it only contains only managed objects, it doesn't benefit from being disposable.

Reply:
Problem was this cache is passed around on UI many times, so I can't track down where are those vague weak references that not letting this item released even though PM UI (even solution) is already closed.
So instead of chasing it we already have set [null](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L821) logic, so add self release of `Packages` it holding by calling [Dispose()](https://github.com/NuGet/NuGet.Client/blob/7ad196d8c3ab6a883e94372b5ba0c54e66ee25ea/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L823) before we set null. So it releases `Packages` without having to find all weak references of cache. Of cause it's not ideal solution but easiest solution without dragged into XAML code (causing that weak references) which much complicated. Most likely making PM UI [out of process](https://github.com/NuGet/Client.Engineering/issues/386) will solve this.